roles/node/tasks/400-binary.yml

```diff
@@ -5,13 +5,25 @@
     _node_binaries: "{{ [{'url': node_binary, 'signature_url': node_binary_signature, 'dst': _node_main_binary_file_name}] }}"
     _node_temp_binary_path: "{{ _node_temp_dir.path }}/bin"
 
+- name: Binary | Check current binary path exists
+  ansible.builtin.stat:
+    path: "{{ _node_binary_path }}"
+  register: _node_binary_path_stat
+
+- name: Post tasks | Remove old binary file
+  ansible.builtin.file:
+    path: "{{ _node_binary_path }}"
+    state: absent
+  when: _node_binary_path_stat.stat.exists and not _node_binary_path_stat.stat.isdir
```

Comment:
can you move this logic after we download new binary in temp dir?
use case:
1. the `node_binary` variable is not correct
2. ansible delete the old binary  (you did not notice it)
3. ansible will fail to download new binary and fail playbook.
4. the node will probably will continue to run even without binary, but after restart systemd will fail to start it.

 
Better move this logic after `Binary | Check current version` task.